### ISSUE TITLE
Verification email when user email is changed

### ISSUE THREAD

**Issue by alexmylonas** [panther-labs/panther#313]
### Describe the bug

After #107 is implemented we 'll allow users to change emails without confirmation/verification
This could potentially lead to account deadlock.

### Steps to reproduce

1. Sign in to platform 
2. Change your email to a one you don't have access or misspell the new email
3. If you forget your password, you can't reset your password

### Expected behavior

A verification email to the new and/or old email before persist the new email

**Comment by 3nvi** [comment]
@austinbyers what's the status on this? Is it still valid based on our recent work on reset-password & verifications?

**Comment by austinbyers** [comment]
It doesn't look like you can change your own email from the profile icon, but admins can still change user emails to anything they want, yes. So it's possible for an admin to change their own email, mistype it, and be locked out. A similar problem happens if you mistype the email when deploying.

I just don't know what we can reasonably do about it. We mark emails as "auto-verified" in Cognito, I have no idea how the user flow would change if we require emails to be verified or how much work it would be to implement.

We can leave this open as something to research someday, or we can close it as low priority and  not something we'll realistically have time to implement. Other admins can fix your email for you, or you can fix it from the users-api / Cognito directly

**Comment by 3nvi** [comment]
I think that we should keep it open. Meanwhile perhaps asking for the user's email 2 times (both during `mage deploy` and within the UI) can perhaps temporarily help reduce the cases of mistypings.

I know for sure that I mistyped my email once during `mage deploy` and I had to go through Cognito and lose some time

**Comment by austinbyers** [comment]
This isn't really a bug, it's a feature - require email verification when updating user emails. As such, I've moved this to product board: https://runpanther.productboard.com/feature-board/1567477-feature-organization/features/5467897/detail

Until then, if you lock yourself out with the wrong email, an admin can reset it, or we have an `invite` opstool which can manually invite a new user for you (which will eventually be superseded by the real API)
